src/scenario/message_media_text_blob.c

```diff
@@ -0,0 +1,164 @@
+#include "message_media_text_blob.h"
+
+#include "core/log.h"
+#include "core/string.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#define MESSAGE_MEDIA_TEXT_BLOB_INITIAL_SIZE (100 * 1024)
+#define MESSAGE_MEDIA_TEXT_BLOB_SIZE_INCREASE_OVERSHOOT (10 * 1024)
+#define MESSAGE_MEDIA_TEXT_BLOB_INITIAL_ENTRIES 100
+#define MESSAGE_MEDIA_TEXT_BLOB_ENTRIES_INCREASE_OVERSHOOT 10
+
+static message_media_text_blob_t message_media_text_blob;
+
+static void resize_text_blob(int needed_space)
+{
+    if (!message_media_text_blob.max_size_text_blob) {
+        int size = MESSAGE_MEDIA_TEXT_BLOB_INITIAL_SIZE * sizeof(uint8_t);
+        message_media_text_blob.text_blob = (uint8_t *) malloc(size);
+        message_media_text_blob.max_size_text_blob = size;
+    }
+    int needed_max_size = message_media_text_blob.size + needed_space;
+    if (message_media_text_blob.max_size_text_blob < needed_max_size) {
+        int size = needed_max_size + MESSAGE_MEDIA_TEXT_BLOB_SIZE_INCREASE_OVERSHOOT;
+        message_media_text_blob.text_blob = (uint8_t *) realloc(message_media_text_blob.text_blob, size);
```

Comment:
Beware. Realloc may fail and in that case not only you'll likely get a crash here, but you'll also get a memory here as the old pointer won't be freed if the new allocation fails.